elizaOS/eliza
# feat: Add GitBook Plugin provider

# Relates to:

N/A


# Risks

Low: Adding new features 
- Adds GitBook documentation querying capability
- No database changes
- No security implications

# Background

## What does this PR do?
Adds a new GitBook documentation plugin that enables:
- Dynamic documentation querying through GitBook API
- Configurable keyword and trigger matching
- Automatic response generation from documentation content

## What kind of change is this?

Features: 
- Adds ability to query GitBook for internal docs
- Adds new client configuration options for GitBook integration
- Implements smart query validation to prevent API spam



# Documentation changes needed?

Yes, my changes require updates to the project documentation:

1. Add new GitBook configuration section to character.json:

```
{
    "clientConfig": {
        "gitbook": {
            "keywords": {
                "projectTerms": ["term1", "term2"],  // Project-specific terms
                "generalQueries": ["custom1", "custom2"]  // Additional query terms
            },
            "documentTriggers": ["help", "docs", "guide"]  // Documentation triggers
        }
    }
}
```

# Testing

## Where should a reviewer start?
Review the plugin implementation in:

plugin-gitbook/src/providers/gitbook.ts
plugin-gitbook/src/types.ts
plugin-gitbook/src/index.ts

Check the character configuration handling

## Detailed testing steps


1. Configure GitBook space ID:
`GITBOOK_SPACE_ID=your_space_id`

2. Test with different configurations:

- Without any gitbook config (should use defaults)
- With partial config (should merge with defaults)
- With full config (should use provided settings)

3. Test query validation:

- General queries
- Project-specific terms
- Document triggers
- Combinations of above


4. Verify response handling:

- Correct documentation retrieval
- Proper response formatting
- Error handling






- Plugin is designed to be configurable but work with sensible defaults
- Implementation focuses on preventing unnecessary API calls while maintaining functionality
- No database changes required
- No special deployment steps needed
